src/Silk.Core.Discord/Commands/General/ClearCommand.cs

```diff
@@ -1,35 +1,76 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Collections.Generic;
 using System.Threading.Tasks;
 using DSharpPlus;
 using DSharpPlus.CommandsNext;
 using DSharpPlus.CommandsNext.Attributes;
 using DSharpPlus.Entities;
 using DSharpPlus.Exceptions;
+using MediatR;
+using Silk.Core.Data.MediatR.Unified.Guilds;
+using Silk.Core.Data.Models;
 using Silk.Core.Discord.Utilities.HelpFormatter;
 
 namespace Silk.Core.Discord.Commands.General
 {
     [Category(Categories.Mod)]
     public class ClearCommand : BaseCommandModule
     {
+        private readonly IMediator _mediator;
+
+        public ClearCommand(IMediator mediator)
+        {
+            _mediator = mediator;
+        }
+
         [Command]
-        [RequireUserPermissions(Permissions.ManageMessages)]
         [Description("Cleans all messages from all users.")]
+        [RequireUserPermissions(Permissions.ManageMessages)]
         public async Task Clear(CommandContext ctx, int messages = 5)
         {
+            GuildConfig guildConfig = await GetOrCreateGuildConfig(ctx);
+            
+            ulong loggingChannelId = guildConfig.LoggingChannel;
+            DiscordChannel? loggingChannel = ctx.Guild.GetChannel(loggingChannelId) ?? ctx.Channel;
```

Comment:
~~use null coalescing operator (`?.`) instead, and~~ log the original embed (x messages cleared) to the channel regardless of logging is setup.